From 043c7166b37f5ab514a88cce621188e6e3f384b3 Mon Sep 17 00:00:00 2001
From: Yong Zhang <yong.zhang@windriver.com>
Date: Thu, 19 Mar 2015 09:51:21 -0400
Subject: [PATCH] MIPS: XLP: irq: mark xlp_pic IRQCHIP_ONESHOT_SAFE

Testing shows that the irq chip is oneshot safe.

And look at the source code, xlp_pic_mask_ack()/xlp_pic_unmask()
must be called on the local cpu; once it's called in irq thread,
it's possible that the irq will leave masked. So mark it
IRQCHIP_ONESHOT_SAFE to resolve the issue.

Signed-off-by: Yong Zhang <yong.zhang@windriver.com>
[NN: this is needed to avoid
 "ata1.00: qc timeout (cmd 0x27)" error at boot]
Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/arch/mips/netlogic/common/irq.c b/arch/mips/netlogic/common/irq.c
index 963fd8b71910..f853177af965 100644
--- a/arch/mips/netlogic/common/irq.c
+++ b/arch/mips/netlogic/common/irq.c
@@ -180,6 +180,7 @@ static struct irq_chip xlp_pic = {
 	.irq_mask_ack	= xlp_pic_mask_ack,
 	.irq_unmask	= xlp_pic_unmask,
 	.irq_set_affinity = xlp_pic_set_affinity,
+	.flags		= IRQCHIP_ONESHOT_SAFE,
 };
 
 static void cpuintr_disable(struct irq_data *d)
-- 
2.1.0

